crates/gen/src/types/class.rs

```diff
@@ -233,6 +239,52 @@ impl Class {
         }))
     }
 
+    fn to_call_factory_tokens(&self) -> TokenStream {
+        let mut tokens = Vec::new();
+
+        if self.default_constructor {
+            let interface_tokens = quote! { ::winrt::IActivationFactory };
+            tokens.push(self.to_named_call_factory("IActivationFactory", &interface_tokens));
+        }
+
+        for interface in &self.interfaces {
+            if interface.kind != InterfaceKind::Statics
+                && interface.kind != InterfaceKind::Composable
+            {
+                continue;
+            }
+
+            let interface_namespace =
+                to_namespace_tokens(&interface.name.namespace, &self.name.namespace);
+
+            let interface_name = format_ident(&interface.name.name);
+            let interface_tokens = quote! { #interface_namespace #interface_name };
+            tokens.push(self.to_named_call_factory(&interface.name.name, &interface_tokens));
+        }
+
+        TokenStream::from_iter(tokens)
+    }
+
+    fn to_named_call_factory(&self, method_name: &str, interface: &TokenStream) -> TokenStream {
+        let self_name = &self.name.tokens;
+        let method_name = format_ident(method_name);
+
+        quote! {
+            #[allow(non_snake_case)]
+            fn #method_name<R, F: FnOnce(&#interface) -> ::winrt::Result<R>>(
+                callback: F,
+            ) -> ::winrt::Result<R> {
+                static mut SHARED: ::winrt::FactoryCache<#self_name, #interface> =
```

Comment:
Unfortunately, Rust doesn't support type-specific statics so we need to generate a function for each class interface to house the static storage.